Spice simulation script

mattddobre opened the pull request.

mattddobre committed:
Add files via upload

mattddobre committed:
Create README.md

mattddobre committed:
Rename component_sim.py to spice_simulation_generator/component_sim.py

mattddobre committed:
Rename global.conf to spice_simulation_generator/global.conf

mattddobre committed:
Rename spice_gen.py to spice_simulation_generator/spice_gen.py

mattddobre committed:
Update README.md

rmanohar commented on spice_simulation_generator/component_sim.py:
```diff
@@ -0,0 +1,33 @@
+#PINS = ["din[0]", "din[1]", "wl[0]", "wl[1]", "go", "sen", "wen", "pchg", "wreq", "Reset", "en1_s", "en2_s", "en1_m", "en2_m", "dout[0]", "dout[1]", "dr", "wc"]
+
+# Model file
+MODEL = "/gpfs/gibbs/pi/manohar/tech/SKY/130/skywater-pdk/libraries/sky130_fd_pr/latest/models/sky130.lib.spice"
```
This is very specific to a particular install. Can we change this to $ACT_HOME/$ACT_TECH/models.sp?

rmanohar commented on spice_simulation_generator/component_sim.py:
```diff
@@ -0,0 +1,33 @@
+# Creates spice deck to simulate a particular component
```
Perhaps add a comment here that this is an example of using the spice_gen.py module.

rmanohar commented on spice_simulation_generator/global.conf:
This should not be committed. This is technology-specific, and should be part of the ACT install.

rmanohar reviewed (changes requested):
I've added a few suggested changes to this PR.

mattddobre committed:
Delete spice_simulation_generator directory

mattddobre committed:
Create README.md

mattddobre committed:
Add files via upload

Includes addition of basic parser, and can now extract pins automatically from dut netlist generated from prs2net.

mattddobre committed:
Rename component_sim.py to spice_simulation_generator/component_sim.py

mattddobre committed:
Rename spice_gen.py to spice_simulation_generator/spice_gen.py
